packages/nx/src/command-line/migrate.ts

```diff
@@ -434,6 +443,80 @@ export class Migrator {
     ]).then((a: { shouldApply: boolean }) => a.shouldApply);
   }
 
+  private getPkgVersion(pkg: string): string {
+    return this.getInstalledPackageVersion(
+      pkg,
+      this.installedPkgVersionOverrides
+    );
+  }
+
+  private normalizePackageGroup(
+    packageName: string,
+    targetVersion: string,
+    packageGroup: PackageGroup
+  ): { package: string; version: string }[] {
+    // Support Migrating to older versions of Nx
+    // Use the packageGroup of the latest version of Nx instead of the one from the target version which could be older.
+    if (
+      packageName === '@nrwl/workspace' &&
+      lt(targetVersion, '14.0.0-beta.0')
+    ) {
+      packageGroup = {
+        '@nrwl/workspace': '*',
+        '@nrwl/angular': '*',
+        '@nrwl/cypress': '*',
+        '@nrwl/devkit': '*',
+        '@nrwl/eslint-plugin-nx': '*',
+        '@nrwl/express': '*',
+        '@nrwl/jest': '*',
+        '@nrwl/linter': '*',
+        '@nrwl/nest': '*',
+        '@nrwl/next': '*',
+        '@nrwl/node': '*',
+        '@nrwl/nx-plugin': '*',
+        '@nrwl/react': '*',
+        '@nrwl/storybook': '*',
+        '@nrwl/web': '*',
+        '@nrwl/js': '*',
+        '@nrwl/cli': '*',
+        '@nrwl/nx-cloud': 'latest',
+        '@nrwl/react-native': '*',
+        '@nrwl/detox': '*',
+        '@nrwl/expo': '*',
+      };
+    }
+
+    if (!packageGroup) {
+      return [];
+    }
+
+    if (!Array.isArray(packageGroup)) {
+      return Object.entries(packageGroup).map(([pkg, version]) => {
+        if (this.installedPkgVersionOverrides[packageName] && version === '*') {
+          this.installedPkgVersionOverrides[pkg] ??=
+            this.installedPkgVersionOverrides[packageName];
+        }
+        return { package: pkg, version };
+      });
+    }
+
+    return packageGroup.map((packageConfig) => {
+      if (this.installedPkgVersionOverrides[packageName]) {
+        if (typeof packageConfig === 'string') {
+          this.installedPkgVersionOverrides[packageConfig] ??=
+            this.installedPkgVersionOverrides[packageName];
+        } else if (packageConfig.version === '*') {
+          this.installedPkgVersionOverrides[packageConfig.package] ??=
+            this.installedPkgVersionOverrides[packageName];
+        }
+      }
+
+      return typeof packageConfig === 'string'
+        ? { package: packageConfig, version: targetVersion }
+        : packageConfig;
+    });
+  }
```

Comment:
I'm moving a lot of this logic into the readMigrationsConfig fn inside https://github.com/nrwl/nx/pull/14648, as a heads up. I'll approve + merge this and rebase 14648 to get it fixed up.

Reply:
Cool, please let me know when you do to have a look and double-check things so nothing gets overlooked on the rebase.